src/js/angular/graphql/controllers/graphql-endpoint-management-view.controller.js

```diff
@@ -86,6 +93,38 @@ function GraphqlEndpointManagementViewCtrl($scope, $location, $interval, $reposi
     // Public methods
     // =========================
 
+    /**
+     * Sets the given endpoint as the default one for the active repository.
+     * @param {GraphqlEndpointInfo} endpointInfo The endpoint to set as default.
+     */
+    $scope.setEndpointAsDefault = (endpointInfo) => {
+        const previousDefaultEndpoint = $scope.selectedDefaultEndpoint;
+        $scope.selectedDefaultEndpoint.default = false;
+        endpointInfo.default = true;
+        $scope.selectedDefaultEndpoint = endpointInfo;
+        $scope.operationInProgress = true;
+        const updateEndpointRequest = endpointInfo.toUpdateEndpointRequest($scope.endpointConfigurationSettings);
+        GraphqlService.editEndpointConfiguration($repositories.getActiveRepository(), endpointInfo.endpointId, updateEndpointRequest)
+            .then(() => {
+                toastr.success(
+                    $translate.instant('graphql.endpoints_management.table.actions.set_as_default.success',
+                        {endpointId: endpointInfo.endpointId})
+                );
+                return loadEndpointsInfo(false);
+            })
+            .catch((error) => {
+                // something wen wrong while setting the default endpoint so we need to revert the changes
```

Comment:
Previously, before sending the request, we set `endpointInfo.default = true`. We may need to revert this setting if an error occurred.

Reply:
Well. I'm doing it. See line 119

Reply:
done